src/Markdig/Extensions/Emoji/EmojiMapping.cs

```diff
@@ -0,0 +1,1784 @@
+// Copyright (c) Alexandre Mutel. All rights reserved.
+// This file is licensed under the BSD-Clause 2 license. 
+// See the license.txt file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Markdig.Helpers;
+
+namespace Markdig.Extensions.Emoji
+{
+    /// <summary>
+    /// An emoji and smiley mapping, to be used by <see cref="EmojiParser"/>.
+    /// </summary>
+    public class EmojiMapping
+    {
+        /// <summary>
+        /// The default emojis and smileys mapping.
+        /// </summary>
+        public static readonly EmojiMapping DefaultEmojiMapping = new EmojiMapping();
+
+        internal CompactPrefixTree<string> EmojiPrefixTree { get; }
+        internal CompactPrefixTree<string> EmojiSmileyPrefixTree { get; }
+        internal char[] EmojiOpeningCharacters { get; }
+        internal char[] EmojiSmileyOpeningCharacters { get; }
+
+        #region Emojis and Smileys
+
+        /// <summary>
+        /// Returns a new instance of the default emoji to unicode dictionary.
+        /// It can be used to create a customized <see cref="EmojiMapping"/>.
+        /// </summary>
+        public static IDictionary<string, string> GetDefaultEmojiToUnicode()
+        {
+            return new Dictionary<string, string>()
```

Comment:
We could hardcode the capacity here (same for smileys below)

Reply:
@mlaily could you add the correct capacities, I will merge the PR after that, thanks!

Reply:
Oh, sorry, I missed that.